### PR TITLE
1.24: kubelet: remove --redirect-container-streaming flag

### PR DESCRIPTION
**Pull request by pacoxu**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
```
--redirect-container-streaming is no longer functional. The flag will be removed in v1.24
```
refer to https://github.com/kubernetes/kubernetes/pull/100901#issuecomment-828846144
The flag will be removed in v1.24.

#### Which issue(s) this PR fixes:
part of https://github.com/kubernetes/enhancements/issues/1558
follow up of ##95935

#### Special notes for your reviewer:
The default behavior (proxy streaming requests through the kubelet) will be the only supported option.

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper.go [hunk 1]
```diff
@@ -82,20 +79,18 @@ var _ utilnet.Dialer = &SpdyRoundTripper{}
 
 // NewRoundTripper creates a new SpdyRoundTripper that will use the specified
 // tlsConfig.
-func NewRoundTripper(tlsConfig *tls.Config, followRedirects, requireSameHostRedirects bool) *SpdyRoundTripper {
+func NewRoundTripper(tlsConfig *tls.Config, requireSameHostRedirects bool) *SpdyRoundTripper {
```
Comment (caesarxuchao) [inline staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper.go:82]: This is a library function. Let's keep the function signature unless necessary.

Also perhaps the follwoRedirects is still useful in other use cases?

File: test/e2e_node/services/kubelet.go [hunk 2]
```diff
@@ -87,11 +87,11 @@ func RunKubelet() {
 
 const (
 	// Ports of different e2e services.
-	kubeletReadOnlyPort = "10255"
+	kubeletHealthCheckPort = "10248"
```
Comment (caesarxuchao) [inline test/e2e_node/services/kubelet.go:90]: Why do we have to change this?

File: pkg/registry/core/pod/rest/subresources.go [hunk 3]
```diff
@@ -193,7 +193,7 @@ func (r *PortForwardREST) Connect(ctx context.Context, name string, opts runtime
 
 func newThrottledUpgradeAwareProxyHandler(location *url.URL, transport http.RoundTripper, wrapTransport, upgradeRequired, interceptRedirects bool, responder rest.Responder) *proxy.UpgradeAwareHandler {
 	handler := proxy.NewUpgradeAwareHandler(location, transport, wrapTransport, upgradeRequired, proxy.NewErrorResponder(responder))
-	handler.InterceptRedirects = interceptRedirects && utilfeature.DefaultFeatureGate.Enabled(genericfeatures.StreamingProxyRedirects)
+	handler.InterceptRedirects = interceptRedirects
```
Comment (caesarxuchao) [inline pkg/registry/core/pod/rest/subresources.go:196]: Shouldn't handler.InterceptRedirects be always false if `StreamingProxyRedirects` is not supported anymore?

**Review by caesarxuchao — COMMENTED** [review]
/assign @tallclair

File: test/e2e_node/services/kubelet.go (hunk 2 above)
Comment (pacoxu) [reply test/e2e_node/services/kubelet.go:90]: My mistake.
#101139 was mistakenly merged into this PR.
The reason can be found in #101139.

I will drop this commit here.

File: pkg/registry/core/pod/rest/subresources.go (hunk 3 above)
Comment (pacoxu) [reply pkg/registry/core/pod/rest/subresources.go:196]: After re-read the KEP, I think you are right. It should be always false here.

File: staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper.go (hunk 1 above)
Comment (pacoxu) [reply staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper.go]: updated

**Comment by tallclair** [comment]
We decided to go with the more conservative timeline after https://github.com/kubernetes/kubernetes/pull/95935 was merged, so this change needs to wait until v1.24 now.

The step we need to take in v1.22 is:
> Default StreamingProxyRedirects to disabled.

(See the top description on https://github.com/kubernetes/enhancements/issues/1558)

**Pull request closed without merging**